lib/Sema/CSDiag.cpp

```diff
@@ -5372,6 +5382,137 @@ bool FailureDiagnosis::diagnoseCallContextualConversionErrors(
+            return checkTuple(tup1, tup2);
+          else
+            return CC_GeneralMismatch;
+        }
+        if (type1->isEqual(type2) || type2->is<GenericTypeParamType>() ||
+            CS.TC.isSubtypeOf(type1, type2, CS.DC))
+          return CC_ExactMatch;
+        return CC_GeneralMismatch;
+      }
+
+      Closeness checkTuple(TupleType *tup1, TupleType *tup2) {
+        if (tup1->getNumElements() != tup2->getNumElements())
+          return CC_GeneralMismatch;
+
+        for (unsigned i = 0, e = tup1->getNumElements(); i < e; i ++) {
+          auto element1 = tup1->getElement(i);
+          auto element2 = tup2->getElement(i);
+          if (element1.hasName()) {
+            if (!element2.hasName() || element1.getName() !=
+                element2.getName())
+              return CC_GeneralMismatch;
+          }
+          if (checkType(element1.getType(),
+                        element2.getType()) != CC_ExactMatch)
+            return CC_GeneralMismatch;
+        }
+        return CC_ExactMatch;
+      }
+    };
+
+    candidates.filterList([&](UncurriedCandidate cand) -> ClosenessPair {
+      auto candFuncType = cand.getUncurriedFunctionType();
+      if (!candFuncType)
+        return {CC_GeneralMismatch, {}};
+
+      auto candParams = candFuncType->getParams();
+      if (params.size() != candParams.size())
+        return {CC_GeneralMismatch, {}};
+
+      auto checker = TypeConvertibleChecker(CS);
+      for (unsigned i = 0, e = params.size(); i < e; i ++) {
+        if (checker.checkType(params[i].getType(),
+                              candParams[i].getType()) == CC_ExactMatch)
+          continue;
+        return {CC_GeneralMismatch, {}};
+      }
+      return {CC_ExactMatch, {}};
+    });
+
+    auto *locator = CS.getConstraintLocator(UDE, ConstraintLocator::Member);
+    auto memberRange = baseExpr->getSourceRange();
+    if (locator)
+      locator = simplifyLocator(CS, locator, memberRange);
+
+    auto baseLoc = baseExpr->getLoc();
+    auto nameLoc = DeclNameLoc(memberRange.Start);
+
+    auto diag = CS.TC.diagnose(baseLoc, diag::did_you_mean_subscript_operator);
+
+    if (candidates.closeness != CC_ExactMatch)
+      return false;
+
+    auto l_bracket = getTokenText(swift::tok::l_square);
+    auto r_bracket = getTokenText(swift::tok::r_square);
+    auto r_paren = getTokenText(swift::tok::r_paren);
+    auto lastArgSym =
+    Lexer::getCharSourceRangeFromSourceRange(CS.TC.Context.SourceMgr,
+                                             argExpr->getEndLoc());
+    diag.fixItReplace(SourceRange(argExpr->getStartLoc()), l_bracket);
+    diag.fixItRemove(nameLoc.getSourceRange());
+    diag.fixItRemove(SourceRange(UDE->getDotLoc()));
+    if (CS.TC.Context.SourceMgr.extractText(lastArgSym) == r_paren)
+      diag.fixItReplace(SourceRange(argExpr->getEndLoc()), r_bracket);
+    else
+      diag.fixItInsertAfter(argExpr->getEndLoc(), r_bracket);
+  }
+  return true;
```

Comment:
Could you please change it so it doesn't return `true` in this case because it didn't really diagnose anything?